### PR TITLE
fix: remap container paths to host paths in job_metadata.json

### LINKED ISSUE #210: Output metadata from running guidance should include the correct/real paths to objects.
Currently our output metadata is written to a file in the output directory called job_metadata.json. It looks like this:
```
{
"protein": "1VME_0.75occA_0.25occB", 
"structure": "/data/inputs/processed/1VME/1VME_single_001_density_input.cif", 
"density": "/data/inputs/occ_sweeps/0.75occA_0.25occB/density_maps/1VME_0.75occA_0.25occB_1.00A.ccp4", "model": "rf3", 
"guidance_type": "pure_guidance", 
"log_path": "/data/results/1VME_0.75occA_0.25occB/rf3_X-RAY_DIFFRACTION/pure_guidance/ens8_gw0.1/run.log", "output_dir": "/data/results/1VME_0.75occA_0.25occB/rf3_X-RAY_DIFFRACTION/pure_guidance/ens8_gw0.1", "partial_diffusion_step": 120, 
"loss_order": 2, 
"resolution": 1.0, 
"device": "cuda:0", 
"gradient_normalization": true, 
"em": false, 
"guidance_start": -1, 
"augmentation": true, 
"align_to_input": true, 
"step_size": 0.1, 
"step_scaler_type": "noisespace", 
"model_checkpoint": "/checkpoints/rf3_foundry_01_24_latest.ckpt", 
"msa_path": null, 
"disable_chiral_features": true, 
"track_chiral_features": false, 
"ensemble_size": 8
}
```
Note that the paths to input files start with `/data/inputs`, `/data/results`, etc... These are paths inside the ephemeral container that runs the ensemble generation jobs. No matter where we run these commands, we'd like the paths to point to the actual permanent files/directories, so that the experiment can be recreated if necessary.

### PR DESCRIPTION
**Pull request by Abdelsalam-Abbas**
Closes #210

## Summary

- When guidance runs inside Docker, `job_metadata.json` recorded container-internal paths (`/data/inputs/...`, `/data/results/...`) instead of host paths, making it impossible to reproduce experiments from metadata alone
- Add `_remap_container_path()` in `GuidanceConfig.as_dict()` that substitutes container path prefixes with host paths using env vars
- Three env vars, checked in specificity order: `SAMPLEWORKS_HOST_INPUT_DIR` (`/data/inputs`), `SAMPLEWORKS_HOST_RESULTS_DIR` (`/data/results`), `SAMPLEWORKS_HOST_DIR` (`/data` catch-all for single mounts)
- Remapping only affects metadata serialization; runtime file I/O is unchanged
- When no env vars are set, behavior is identical to before

## Test plan

- [x] Unit tests for `_remap_container_path`: no-op without env vars, split mount remapping, catch-all remapping, specificity priority, checkpoint passthrough, trailing slash normalization, exact prefix match, empty env var handling
- [x] Integration tests for `as_dict()`: split mounts, catch-all, and no-env-var backward compat
- [x] `pixi run -e boltz-dev tests -- tests/utils/test_guidance_script_arguments.py -v` (39 passed)

## Summary by CodeRabbit

* **New Features**
  * Added environment variables to record host paths in job metadata: SAMPLEWORKS_HOST_INPUT_DIR, SAMPLEWORKS_HOST_RESULTS_DIR, or single SAMPLEWORKS_HOST_DIR.

* **Chores**
  * Docker run commands updated to pass these host-path environment variables into containers.

* **Documentation**
  * Help/usage output gained a METADATA HOST PATH REMAPPING section with examples.

* **Tests**
  * Added tests covering remapping behavior, precedence, normalization, and unaffected paths.

### CODE REVIEW

File: tests/utils/test_guidance_script_arguments.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
 import pytest
 from sampleworks.utils.guidance_constants import GuidanceType, StructurePredictor
 from sampleworks.utils.guidance_script_arguments import (
+    _remap_container_path,
     get_checkpoint,
```
[the remark this answers is not in the document]
Comment (Abdelsalam-Abbas) [reply tests/utils/test_guidance_script_arguments.py:13]: _remap_container_path is a self-contained function with complex edge cases (prefix specificity, slash normalization, root path, whitespace). It's the kind of function that deserves direct unit tests bacause
*  edge cases are about the function's logic, not about how it's called
* The integration tests would just be testing the same logic with 4x the boilerplate
* If someone inlines the function into as_dict(), updating the test imports is trivial

**Review by k-chrispens — APPROVED** [review]
LGTM!
